src/rounded_rect.rs

```diff
@@ -0,0 +1,302 @@
+        self.rect.center()
+    }
+
+    /// Take absolute value of width, height and radius.
+    ///
+    /// The resulting rect has the same extents as the original, but is
+    /// guaranteed to have non-negative properties.
+    #[inline]
+    pub fn abs(&self) -> RoundedRect {
+        RoundedRect {
+            rect: self.rect.abs(),
+            radius: self.radius(),
+        }
+    }
+}
+
+#[doc(hidden)]
+pub struct RoundedRectPathIter {
+    idx: usize,
+    rect: RectPathIter,
+    arcs: [ArcAppendIter; 4],
+}
+
+impl Shape for RoundedRect {
+    type BezPathIter = RoundedRectPathIter;
+
+    fn to_bez_path(&self, tolerance: f64) -> RoundedRectPathIter {
+        let radius = self.radius();
+        let radii = Vec2 {
+            x: self.radius,
+            y: self.radius,
+        };
+
+        let build_arc_iter = |i, center| {
+            let arc = Arc {
+                center,
+                radii,
+                start_angle: FRAC_PI_2 * i as f64,
+                sweep_angle: FRAC_PI_2,
+                x_rotation: 0.0,
+            };
+            arc.append_iter(tolerance)
+        };
+
+        // Note: order follows the rectangle path iterator.
+        let arcs = [
+            build_arc_iter(
+                2,
+                Point {
+                    x: self.rect.x0 + radius,
+                    y: self.rect.y0 + radius,
+                },
+            ),
+            build_arc_iter(
+                3,
+                Point {
+                    x: self.rect.x1 - radius,
+                    y: self.rect.y0 + radius,
+                },
+            ),
+            build_arc_iter(
+                0,
+                Point {
+                    x: self.rect.x1 - radius,
+                    y: self.rect.y1 - radius,
+                },
+            ),
+            build_arc_iter(
+                1,
+                Point {
+                    x: self.rect.x0 + radius,
+                    y: self.rect.y1 - radius,
+                },
+            ),
+        ];
+
+        let rect = RectPathIter {
+            rect: self.rect,
+            ix: 0,
+            radius: self.radius,
+        };
+
+        RoundedRectPathIter { idx: 0, rect, arcs }
+    }
+
+    #[inline]
+    fn area(&self) -> f64 {
+        let radius = self.radius();
+        self.rect.area() - (4.0 - PI) * radius * radius
+    }
+
+    #[inline]
+    fn perimeter(&self, _accuracy: f64) -> f64 {
+        let radius = self.radius();
+        2.0 * (self.width().abs() + self.height().abs()) - 8.0 * radius + 2.0 * PI * radius
+    }
+
+    #[inline]
+    fn winding(&self, pt: Point) -> i32 {
+        let radius = self.radius();
+        let inside_half_width = (self.width() / 2.0 - self.radius()).max(0.0);
+        let inside_half_height = (self.height() / 2.0 - self.radius()).max(0.0);
+
+        // project point out of the inner rectangle (positive quadrant)
+        let px = (pt.x.abs() - inside_half_width).max(0.0);
```

Comment:
I think you need to be subtracting the center from `pt` first; then I believe the logic is probably correct, but haven't validated it carefully.